contract/src/offer-up.contract.js

```diff
@@ -107,15 +107,11 @@ export const start = async zcf => {
       Fail`max ${q(maxItems)} items allowed: ${q(want.Items)}`;
 
     const newItems = itemMint.mintGains(want);
-    atomicRearrange(
-      zcf,
-      harden([
-        // price from buyer to proceeds
-        [buyerSeat, proceeds, { Price: tradePrice }],
-        // new items to buyer
-        [newItems, buyerSeat, want],
-      ]),
-    );
+    const charge = { Price: tradePrice };
+    atomicRearrange(zcf, [
+      { fromSeat: buyerSeat, toSeat: proceeds, fromAmounts: charge },
```

Comment:
Don't bake the types into the property names ("bad Hungarian").  So perhaps
'{ from: buyerSeat, to: proceeds, amounts: charge, mappedTo: zot },'

where `mappedTo: zot` is the optional amountKeywordRecord to rearrange the amounts to.

Reply:
Just for the record, I find a mild amount of Hungarian notation better that both none and too much. 

That said, I like the names @dtribble suggests here. `mappedTo` took awhile to appreciate, but it grew on me.

Reply:
Please restore the `harden([`

Reply:
> Don't bake the types into the property names

Those names came from the existing type declaration. I suppose the existing type is a tuple, where the names don't show up at runtime. ok, done: 2c085f5

Reply:
> Please restore the `harden([`

First I'd like to understand why @dtribble thinks it's not called for.

I think I understand why @erights is asking to put it back:

our [docs](https://docs.agoric.com/guides/js-programming/hardened-js.html#defensive-objects-with-harden) say:
> It's important to harden() an object before exposing the object by returning it or passing it to some other function.

That's a friendly version of the more precise text from the [Jessie readme](https://github.com/endojs/Jessie):
> in Jessie all objects made by literal expressions (object literals, array literals, the many forms of function literals) must be tamper-proofed with harden before they can be aliased or escape from their static context of origin.

Reply:
to me, `harden` is not called for because the arg is an explicit literal, so we cannot have any ability to mutate it, and cannot detect/care if the callee mutates it. If I had a lint rule, I would want that scenario to not be an issue. I'm generally in favor of APIs that harden their args so that the burden is not on the caller, but you guys get to figure that out :).

Reply:
> Just for the record, I find a mild amount of Hungarian notation better that none, and better than too much.
> 
> That said, I like the names @dtribble suggests here. `mappedTo` took awhile to appreciate, but it grew on me.

I also like a mild amount of it, and don't have any objection to "good Hungarian" where it's documenting the role of value, not it's type.  The confusion here is because some have it and some don't.  With naming (e.g.., from/to) there's enough new orientation that it matters less.  The name `proceeds` is very subtle for such a central thing in the contract.

Reply:
> but you guys get to figure that out :).

I am trying to shift us from using objects as an intra-vat defensive boundary to use exo interfaces as a defensive boundary. The exo interfaces certainly are a good defensive boundary in many ways. But their not yet good enough to complete the shift because of promises that can resolve later to things that violate the boundary. @michaelfig and I talked about a progression of promise-likes that would solve the promise whole for this purpose. AFAICT, vows are successfully going in this direction. `M.await` even gives us a safe way to handle arguments that remain promises rather than vows.

I look forward to having the time to get back to these problems. ;)
In the meantime, don't drop `harden`.